### ISSUE TITLE
Install error (MacOS Monterey)

### ISSUE THREAD

**Issue by YingMa0107** [YingMa1993/CARD#8]
Hello @YingMa0107  you did a great job in developing this package !!

 I tried installing the package and I am getting error in installation. I am using Mac with M1 chip and macOS Monterey (12.3.1). The error message and the session info can be found below. Thank you in advance for your help.

_** using staged installation
   ** libs
   clang++ -mmacosx-version-min=10.13 -std=gnu++11 -I"/Library/Frameworks/R.framework/Resources/include" -DNDEBUG  -I'/Library/Frameworks/R.framework/Versions/4.1/Resources/library/Rcpp/include' -I'/Library/Frameworks/R.framework/Versions/4.1/Resources/library/RcppArmadillo/include' -I/usr/local/include    -fPIC  -Wall -g -O2  -Wall -pedantic -fdiagnostics-color=always -c CARDfree.cpp -o CARDfree.o
   clang++ -mmacosx-version-min=10.13 -std=gnu++11 -I"/Library/Frameworks/R.framework/Resources/include" -DNDEBUG  -I'/Library/Frameworks/R.framework/Versions/4.1/Resources/library/Rcpp/include' -I'/Library/Frameworks/R.framework/Versions/4.1/Resources/library/RcppArmadillo/include' -I/usr/local/include    -fPIC  -Wall -g -O2  -Wall -pedantic -fdiagnostics-color=always -c CARDref.cpp -o CARDref.o
   clang++ -mmacosx-version-min=10.13 -std=gnu++11 -I"/Library/Frameworks/R.framework/Resources/include" -DNDEBUG  -I'/Library/Frameworks/R.framework/Versions/4.1/Resources/library/Rcpp/include' -I'/Library/Frameworks/R.framework/Versions/4.1/Resources/library/RcppArmadillo/include' -I/usr/local/include    -fPIC  -Wall -g -O2  -Wall -pedantic -fdiagnostics-color=always -c RcppExports.cpp -o RcppExports.o
   clang++ -mmacosx-version-min=10.13 -std=gnu++11 -dynamiclib -Wl,-headerpad_max_install_names -undefined dynamic_lookup -single_module -multiply_defined suppress -L/Library/Frameworks/R.framework/Resources/lib -L/usr/local/lib -o CARD.so CARDfree.o CARDref.o RcppExports.o -L/Library/Frameworks/R.framework/Resources/lib -lRlapack -L/Library/Frameworks/R.framework/Resources/lib -lRblas -L/usr/local/gfortran/lib/gcc/x86_64-apple-darwin18/8.2.0 -L/usr/local/gfortran/lib -lgfortran -lquadmath -lm -F/Library/Frameworks/R.framework/.. -framework R -Wl,-framework -Wl,CoreFoundation
   ld: warning: directory not found for option '-L/usr/local/gfortran/lib/gcc/x86_64-apple-darwin18/8.2.0'
   ld: warning: directory not found for option '-L/usr/local/gfortran/lib'
   ld: library not found for -lquadmath
   clang: error: linker command failed with exit code 1 (use -v to see invocation)
   make: *** [CARD.so] Error 1
   ERROR: compilation failed for package ‘CARD’
─  removing ‘/private/var/folders/16/2qzgz8cd2bl65zhgv1x6sy0c0000gn/T/Rtmprukk9k/Rinst102f577830c0c/CARD’
         -----------------------------------
   ERROR: package installation failed
Error: Failed to install 'CARD' from GitHub:
  System command 'R' failed, exit status: 1, stdout & stderr were printed
> sessionInfo()
R version 4.1.2 (2021-11-01)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Monterey 12.3.1

Matrix products: default
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
 [1] pillar_1.7.0      compiler_4.1.2    prettyunits_1.1.1 remotes_2.4.2    
 [5] tools_4.1.2       testthat_3.1.2    pkgbuild_1.3.1    pkgload_1.2.4    
 [9] memoise_2.0.1     lifecycle_1.0.1   tibble_3.1.7      pkgconfig_2.0.3  
[13] rlang_1.0.2       rstudioapi_0.13   DBI_1.1.2         cli_3.3.0        
[17] curl_4.3.2        fastmap_1.1.0     withr_2.4.3       dplyr_1.0.9      
[21] generics_0.1.2    desc_1.4.0        fs_1.5.2          vctrs_0.4.1      
[25] devtools_2.4.3    rprojroot_2.0.2   tidyselect_1.1.1  glue_1.6.1       
[29] R6_2.5.1          processx_3.5.2    fansi_1.0.2       sessioninfo_1.2.2
[33] tidyr_1.2.0       purrr_0.3.4       callr_3.7.0       magrittr_2.0.2   
[37] ps_1.6.0          ellipsis_0.3.2    usethis_2.1.5     assertthat_0.2.1 
[41] utf8_1.2.2        cachem_1.0.6      crayon_1.4.2      brio_1.1.3_

_Originally posted by @utkarsh0493 in https://github.com/YingMa0107/CARD/issues/4#issuecomment-1122746949_

**Comment by YingMa0107** [comment]
Hi @utkarsh0493, 

Thank you very much for your interest in our package and thank you for providing a detailed description of your issue! Based on your output, it seems that there is no gfortran or gcc. Could you please check if you have installed the gfortran or gcc compiler? You can open a terminal on your MacBook and type gcc --version, it should have some output, for example, in my MacOS system:  

```
gcc (GCC) 12.1.0
Copyright (C) 2022 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

If not, please install both gfortran and gcc. I was trying to reproduce your error and I updated my available MacOS system to Montery 12.3.1, then I followed this tutorial to reinstall Xcode, gfortran and gcc compiler, and then installed CARD successfully. The tutorial is here: [http://yiqingxu.org/public/BigSurError.pdf](http://yiqingxu.org/public/BigSurError.pdf), and I found it is very useful and easy to follow. 

Let me know if it helps or not!

**Comment by utkarsh0493** [comment]
Hi @YingMa0107 

I checked for gcc on my machine, it was there (older version though).
After following your tutorial document (I should admit, it is super handy and useful for this gfortran issues), I updated the gfortran to latest version. And found out that the path to gfortran was missing from the PATH variable, updating it with correct information solved my issue.
